Use path helpers consistently

dometto opened the pull request:
Found a couple of instances where the javascript was not using path helpers. I did some extensive grepping, including for `'/`, `baseUrl`, and all the currently existing route names, and moved to using the path helpers.

dometto commented:
One of the things I found was `gollum.editor.js.erb:12225`:
`          $.post('/wiki/help?_method=delete');`
I honestly have no idea what that's supposed to do. Do you think we can remove it?

dometto commented:
Also, remind me again while we still have a `fileview.scss`?

dometto commented:
Just realized this: the current setup would make it impossible to use `base_path` with static assets, because the paths to routers in the assets are generated at asset precompilation time. So we should, after all, prefix `baseUrl` in the JS. We can do this cleaner than previously, though, by having a JS function that prefixes `baseUrl` and consistently using that wherever we're now using a path helper.

bartkamphorst commented:
> So we should, after all, prefix baseUrl in the JS.

Good observation. I take it you no longer need my review for this PR then, right?

dometto commented:
I'll make the required changes and then ask for a review!

dometto commented on lib/gollum/public/gollum/javascript/editor/gollum.editor.js.erb:
```diff
@@ -405,43 +398,6 @@
       return null;
     },
 
-
```
Turns out the `loadFor` method wasn't doing anything anymore, since we're requiring all the language defintions in `app.js`, so was able to remove this block.

bartkamphorst replied:
👍

dometto commented:
@bartkamphorst I've now added a `routePath` helper method to the JS which borrows the route definitions from the route helpers. Using `routePath` in the JS thus allows us to easily generate urls, while prefixing `baseUrl` on the go (if necessary).

I've also:
* addressed #1355 
* caught a bug with the file deletion button in the pages view when using basepath (please check)
* moved the precompilation task to the rakefile

I've tested with static and dynamic assets, with and without basefile, and wasn't able to catch any bugs. But please inspect. :)

bartkamphorst commented:
> $.post('/wiki/help?_method=delete');  I honestly have no idea what that's supposed to do. Do you think we can remove it?

No, I believe this is used for showing or removing the "help" functionality in the editor.

bartkamphorst commented:
> Also, remind me again while we still have a fileview.scss?

Only as a reminder / example for how to improve some aspects of the Overview page.

bartkamphorst commented:
> I've now added a routePath helper method to the JS which borrows the route definitions from the ruby route helpers.

Elegantly done. 👍

bartkamphorst commented:
> caught a bug with the file deletion button in the pages view when using basepath

Your change seems sensible. I'll test everything out tomorrow.

bartkamphorst commented on lib/gollum/public/gollum/javascript/gollum.js.erb:
```diff
@@ -1,4 +1,27 @@
+  var routes = $.parseJSON('<%= routes_to_json %>')
+  path = routes[name]
+  if (baseUrl == undefined ) {
+    console.log('Gollum error: baseUrl undefined')
```
When does this happen (baseUrl == undefined)? And if it happens, does it break functionality?

dometto replied:
It would happen if we somehow forget to set the `baseUrl` variable in the page template. I thought it would be useful to add the debug message in case that ever happens.

bartkamphorst replied:
👍

bartkamphorst commented on lib/gollum/public/gollum/javascript/editor/gollum.editor.js.erb:
```diff
@@ -405,43 +398,6 @@
 
-
-    /**
-     *  loadFor
```
Are we sure that this is deprecated and no longer used?

dometto replied:
Define "sure" ;)

bartkamphorst replied:
OK.

bartkamphorst commented on Rakefile:
```diff
@@ -177,3 +177,23 @@ task :validate do
+ require 'sprockets-helpers'
+ require 'sass'
+  env = Precious::Assets.sprockets
+  manifest = Sprockets::Manifest.new(env, './public/assets')
```
It would be nice to add an option here to compile the assets to a custom location.

dometto replied:
👍

bartkamphorst commented:
Actually, did you resolve https://github.com/gollum/gollum/issues/1356 in this PR?
